### DISCUSSION TITLE
Is depvi-client user/index configuration by file possible?

### DISCUSSION THREAD

**Discussion by silvanbrummen** [devpi/devpi#877] [Q&A]
Heya!

At the moment I'm trying to set up a devpi server, running in a docker container. While reading the documentation I noticed that the preferred way of setting the configuration is by a file and the --configfile option. 

However, I'm wondering if it's also possible to configure user and index setting by file. Normally this should be done by the devpi-client, but I would like to do this file-based as well. Is this possible at the moment? And if yes, how should this be done?

Thanks in advance,
Sil

**Comment by fschulze** [comment]
The only process going in that direction is export/import. I'm curious why you would want that though. I see Devpi more like a Database with persistent data.
If you go the import route, you might want to replace the UID in the json file, otherwise all your instances will have the same UID.

**Comment by jamesdbrock** [comment]
I'm working on creating a  devpi-server with NixOS. I would also like to be able to declare the full specification for the server in a file.

Right now my approach is to use all the interactive commands like `devpi user --create` to configure the server, and then `devpi-export` the server state to a `dataindex.json` file which I can check into the Git repo for the NixOS server. I don't think this is going to work very well though. For one thing, suppose I want to add another index some day? I can't import a new `dataindex.json` without first deleting the entire server, can I?

**Reply by stevedrees** [reply]
Are you not utilizing volumes to make the sqllite db persistent? That is the route I would go here.

**Comment by fschulze** [comment]
@jamesdbrock how do you handle databases like Postgresql? Essentially devpi-server is the same. You have an init command and then a client of some sort which changes the database. I don't know a way to do something like you want with any DB. If there is something in NixOS that does what you want for Postgresql, then I'm pretty certain there is some wrapper in the background where a client call to the database is made.
